cpu/atmega_common/periph/gpio.c

```diff
@@ -237,61 +293,114 @@ void gpio_write(gpio_t pin, int value)
     }
 }
 
+static inline void pcint_handler(uint8_t port_num, uint8_t pin_num)
+{
+    int gpio_state = gpio_read( GPIO_PIN(port_num, pin_num) );
+    gpio_flank_t flank = pcint_flank[ port_num * 8 + pin_num ];
+
+    if (flank == GPIO_BOTH || (gpio_state && flank == GPIO_RISING) || (!gpio_state && flank == GPIO_FALLING)) {
```

Comment:
You can't assume which the pin triggered the interrupt. You have to keep track of the previous state of the port state.

Reply:
True. Breaks for more than one pin.